src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -83,6 +87,10 @@ public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterC
             player.send(new RestoreFullHealthEvent(player));
             Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
             player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
+            Optional<Prefab> prefab = Assets.getPrefab("inventory");
+            StartingInventoryComponent startingInventory = prefab.get().getComponent(StartingInventoryComponent.class);
```

Comment:
We resolve and load the asset every time a player dies. As the respawn inventory does not change, you could just initialize this once for the system and then reuse the value every time you reset the inventory.